### PR TITLE
Prevent crash when no fingerprints enrolled although BIOMETRIC_SUCCESS was returned

### PR DESCRIPTION
**Pull request by biafra23**
Handles #1511

Changes proposed in this pull request:
- catch IllegalStateException
- disable biometrics

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt [hunk 1]
```diff
@@ -84,7 +85,14 @@ class RepeatPasscodeFragment : BaseViewBindingFragment<FragmentPasscodeBinding>(
                             },
                             confirmCallback = { dialog ->
                                 viewModel.enableBiometry()
-                                viewModel.encryptPasscodeWithBiometricKey(viewAction.passcode)
+                                try {
+                                    //Some device still throw an IllegalStateException
+                                    // (due to no fingerprints enrolled) even though they return BIOMETRIC_SUCCESS
+                                    viewModel.encryptPasscodeWithBiometricKey(viewAction.passcode)
+                                } catch (e: Exception) {
+                                    settingsHandler.useBiometrics = false
```
Comment (elgatovital) [inline app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt:93]: let's add disableBiometry to viewModel and set useBiometrics to false there, like it is set to true in enableBiometry()

File: app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt [hunk 2]
```diff
@@ -84,7 +85,14 @@ class RepeatPasscodeFragment : BaseViewBindingFragment<FragmentPasscodeBinding>(
                             },
                             confirmCallback = { dialog ->
                                 viewModel.enableBiometry()
```
Comment (elgatovital) [inline app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt:87]: I would place this call after viewModel.encryptPasscodeWithBiometricKey. This sequence seems more correct to me. Biometric is enabled only after all necessary work is done.

Comment (biafra23) [reply app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt:87]: This way `settingsHandler.useBiometrics = false` should not be necessary

File: app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt [hunk 3]
```diff
@@ -83,8 +84,14 @@ class RepeatPasscodeFragment : BaseViewBindingFragment<FragmentPasscodeBinding>(
                                 dismissCreatePasscodeFragment()
                             },
                             confirmCallback = { dialog ->
-                                viewModel.enableBiometry()
-                                viewModel.encryptPasscodeWithBiometricKey(viewAction.passcode)
+                                try {
+                                    //Some device still throw an IllegalStateException
+                                    // (due to no fingerprints enrolled) even though they return BIOMETRIC_SUCCESS
+                                    viewModel.encryptPasscodeWithBiometricKey(viewAction.passcode)
```
Comment (elgatovital) [inline app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt:90]: encryptPasscodeWithBiometry and enableBiometry are always used together => the whole try catch block can and should be moved to a viewModel.  enableBiometry method could not only set the setting but also do the biometric key preparation work.

**Pull request merged**